src/core/server/saved_objects/permission_control/acl.ts

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { PrincipalType } from '../../../../core/utils/constants';
+
+export interface Principals {
+  users?: string[];
+  groups?: string[];
+}
+
+export type Permissions = Partial<Record<string, Principals>>;
+
+const addPermissionToPrincipals = (
```

Comment:
Based on the implementation, there is no permission related. i would recommend use `addToPrincipals` as method name.